### ISSUE TITLE
No prompts inputted into ExtendSAM

### ISSUE THREAD

**Issue by KainingYing** [ziqi-jin/finetune-anything#22]
Thank you very much for your meaningful work. I have briefly debugged the training of semantic segmentation and found that [no prompt](https://github.com/ziqi-jin/finetune-anything/blob/1264f1d386dc9c3820aec66d112447836911d007/extend_sam/extend_sam.py#L23) inputs were sent to ExtendSAM. My understanding is that it may be appropriate to use 32x32 grid points as a prompt and classify the mask. Or perhaps I have misunderstood the purpose of your prompt inputs, and I'm curious about why you used them.

**Comment by ziqi-jin** [comment]
Yeah, you are very patient, the default SAM uses 32*32 points as prompt, and the FA is designed to input prompt through PromptEncoderAdapter, But I did not design it yet. So now Extend-SAM’s prompt input is ‘None’, don’t worry, I will design PromptEncoderAdapter later. 

Now the performance of fine-tuned Extend-SAM on VOC has been acceptable. After adding the prompt part will be better, please wait a moment.
If you have any idea to design the PromptEncoderAdapter, please submit PR to me, thanks for your attention.

**Comment by KainingYing** [comment]
Why `Now the performance of fine-tuned Extend-SAM on VOC has been acceptable`?
To the best of my knowledge, `No prompts = No outputs`.  In this version of FA, the output of mask tokens are useless. Am I right?

**Comment by ziqi-jin** [comment]
FA support customized MaskDecoderAdapter, and the SemSAM class is an extend-SAM model whose MaskDecoderHead was modified. When the prompt is None, it will output an output, the shape of the output is the standard shape for the semantic segmentation task. The design of MaskDecoderHead in SemSAM is for class-ware mask predictions. The design of   SemSAM may not be very suitable, it is an example for users to design their own Extend-SAM models. Scientific research requires a lot of experimentation and thinking, Scientific research requires a lot of experiments and thinking, we are more inclined to provide a tool to help realize their ideas.

For the prompt part, I will seriously consider your suggestion and try to design an easy-to-use prompt interface.

by the way, if you have a good idea, you are welcome to submit PR to this repository.

**Comment by KainingYing** [comment]
Thanks for your kind reply. I think the MaskDecoder in SemSAM is not prompt-driven anymore. The finetune processing only make a new task-specific head. This paper `(How to Efficiently Adapt Large Segmentation Model(SAM) to Medical Images)` did a similar thing.

**Comment by ziqi-jin** [comment]
> Thanks for your kind reply. I think the MaskDecoder in SemSAM is not prompt-driven anymore. The finetune processing only make a new task-specific head. This paper `(How to Efficiently Adapt Large Segmentation Model(SAM) to Medical Images)` did a similar thing.

Thanks for your attention,

Yes, The SamSAM has no prompt, it is hard to design a model to output class-ware semantic mask, and this repository is not finished. For the semantic segmentation task, if we add different kinds of prompt, the model needs a lot of post-processing and need to merge similar mask, so I didn't add a prompt for semantic segmentation now. If we fix processing for a specific task, the users will lose the freedom to design their own idea. If the research community proposes a better solution, I will use it.

If I support the Instance segmentation or detection, I think the prompt processing will be easier to design.
